discovery-provider/src/tasks/index_user_bank_backfill.py

```diff
@@ -0,0 +1,570 @@
+# Message formatted as follows:
+# EthereumAddress = [214, 237, 135, 129, 143, 240, 221, 138, 97, 84, 199, 236, 234, 175, 81, 23, 114, 209, 118, 39]
+def parse_eth_address_from_msg(msg: str):
+    res = re.findall(r"\[.*?\]", msg)
+    # Remove brackets
+    inner_res = res[0][1:-1]
+    # Convert to public key hex for ethereum
+    arr_val = [int(s) for s in inner_res.split(",")]
+    public_key_bytes = bytes(arr_val)
+    public_key = public_key_bytes.hex()
+    public_key_str = f"0x{public_key}"
+    return public_key_str, public_key_bytes
+
+
+# Return highest user bank slot that has been processed
+def get_highest_user_bank_tx_slot(session: Session):
+    slot = MIN_SLOT
+    tx_query = (
+        session.query(UserBankBackfillTx.slot).order_by(desc(UserBankBackfillTx.slot))
+    ).first()
+    if tx_query:
+        slot = tx_query[0]
+    return slot
+
+
+# Cache the latest value committed to DB in redis
+# Used for quick retrieval in health check
+def cache_latest_sol_user_bank_db_tx(redis: Redis, tx):
+    cache_latest_sol_db_tx(redis, latest_sol_user_bank_backfill_db_tx_key, tx)
+
+
+# Query a tx signature and confirm its existence
+def get_tx_in_db(session: Session, tx_sig: str) -> bool:
+    exists = False
+    tx_sig_db_count = (
+        session.query(UserBankBackfillTx).filter(UserBankBackfillTx.signature == tx_sig)
+    ).count()
+    exists = tx_sig_db_count > 0
+    return exists
+
+
+def refresh_user_balances(session: Session, redis: Redis, accts=List[str]):
+    results = (
+        session.query(User.user_id, UserBankAccount.bank_account)
+        .join(
+            UserBankAccount,
+            and_(
+                UserBankAccount.bank_account.in_(accts),
+                UserBankAccount.ethereum_address == User.wallet,
+            ),
+        )
+        .filter(User.is_current == True)
+        .all()
+    )
+    return results
+
+
+create_token_account_instr: List[InstructionFormat] = [
+    {"name": "eth_address", "type": SolanaInstructionType.EthereumAddress},
+]
+
+
+def process_transfer_instruction(
+    session: Session,
+    redis: Redis,
+    instruction: TransactionMessageInstruction,
+    account_keys: List[str],
+    meta: ResultMeta,
+    tx_sig: str,
+    slot: int,
+    timestamp: datetime.datetime,
+):
+    # The transaction might list sender/receiver in a different order in the pubKeys.
+    # The "accounts" field of the instruction has the mapping of accounts to pubKey index
+    sender_index = instruction["accounts"][TRANSFER_SENDER_ACCOUNT_INDEX]
+    receiver_index = instruction["accounts"][TRANSFER_RECEIVER_ACCOUNT_INDEX]
+    sender_account = account_keys[sender_index]
+    receiver_account = account_keys[receiver_index]
+    # Accounts to refresh balance
+    logger.info(
+        f"index_user_bank_backfill.py | Balance refresh accounts: {sender_account}, {receiver_account}"
+    )
+    user_id_accounts = refresh_user_balances(
+        session, redis, [sender_account, receiver_account]
+    )
```

Comment:
not sure if we need to do this in a backfill job

Reply:
I removed the code that enqueues the user_ids to redis for refresh but I'll rename the fn so it's clearer.